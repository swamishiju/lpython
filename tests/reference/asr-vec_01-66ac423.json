{
    "basename": "asr-vec_01-66ac423",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/vec_01.py",
    "infile_hash": "f85ca108780c53c54878d119822d56fb834cf4b5121511cbaca2c2fe",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-vec_01-66ac423.stdout",
    "stdout_hash": "27f32a4a36f9cad1cac67e1ccba9a98ef01b9dcad4ec21fb037c0cbb",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}