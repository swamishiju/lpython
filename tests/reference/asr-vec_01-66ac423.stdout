(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        229
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [loop_vec]
                                    []
                                    [(SubroutineCall
                                        2 loop_vec
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            loop_vec:
                                (Function
                                    (SymbolTable
                                        225
                                        {
                                            a:
                                                (Variable
                                                    225
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 9216 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    225
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 9216 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    225
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    loop_vec
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 225 a)
                                        (ArrayBroadcast
                                            (ArrayItem
                                                (Var 225 a)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayConstant
                                                4
                                                [9216]
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 1 (Integer 4) Decimal)
                                                    (IntegerConstant 1 (Integer 4) Decimal))]
                                                    FixedSizeArray
                                                )
                                                ColMajor
                                            )
                                            (Array
                                                (Real 8)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 9216 (Integer 4) Decimal))]
                                                FixedSizeArray
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 225 b)
                                        (ArrayBroadcast
                                            (ArrayItem
                                                (Var 225 b)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayConstant
                                                4
                                                [9216]
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 1 (Integer 4) Decimal)
                                                    (IntegerConstant 1 (Integer 4) Decimal))]
                                                    FixedSizeArray
                                                )
                                                ColMajor
                                            )
                                            (Array
                                                (Real 8)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 9216 (Integer 4) Decimal))]
                                                FixedSizeArray
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 225 i)
                                        (IntegerConstant 0 (Integer 4) Decimal)
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4) Decimal)
                                            Sub
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4) Decimal)
                                        )
                                        (IntegerConstant 1 (Integer 4) Decimal))
                                        [(Assignment
                                            (ArrayItem
                                                (Var 225 b)
                                                [(()
                                                (Var 225 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (RealConstant
                                                5.000000
                                                (Real 8)
                                            )
                                            ()
                                            .false.
                                        )]
                                        []
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 225 i)
                                        (IntegerConstant 0 (Integer 4) Decimal)
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4) Decimal)
                                            Sub
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4) Decimal)
                                        )
                                        (IntegerConstant 1 (Integer 4) Decimal))
                                        [(Assignment
                                            (ArrayItem
                                                (Var 225 a)
                                                [(()
                                                (Var 225 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayItem
                                                (Var 225 b)
                                                [(()
                                                (Var 225 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            ()
                                            .false.
                                        )]
                                        []
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 225 i)
                                        (IntegerConstant 0 (Integer 4) Decimal)
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4) Decimal)
                                            Sub
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4) Decimal)
                                        )
                                        (IntegerConstant 1 (Integer 4) Decimal))
                                        [(Assert
                                            (RealCompare
                                                (ArrayItem
                                                    (Var 225 a)
                                                    [(()
                                                    (Var 225 i)
                                                    ())]
                                                    (Real 8)
                                                    RowMajor
                                                    ()
                                                )
                                                Eq
                                                (RealConstant
                                                    5.000000
                                                    (Real 8)
                                                )
                                                (Logical 4)
                                                ()
                                            )
                                            ()
                                        )]
                                        []
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    ()
                    [numpy]
                    .false.
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        230
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    230
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        230 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
